services/index/internal/service/node.go

```diff
@@ -50,35 +50,80 @@ func NewNodeService(
 	}
 }
 
-// SetNodeValid sets a node as valid.
+// SetNodeValid sets a node as valid, updates its status, and indexes it in the
+// repositories.
 func (s *nodeService) SetNodeValid(node *model.Node) error {
+	// Generate and set node ID using SHA256 hash of the ProfileURL.
 	node.ID = cryptoutil.ComputeSHA256(node.ProfileURL)
+	// Set node status to validated and reset failure reasons.
 	node.SetStatusValidated()
 	node.ResetFailureReasons()
 
+	// Retrieve the old node from the repository.
+	oldNode, err := s.mongoRepo.GetByID(node.ID)
+	if err != nil && !errors.As(err, &index.NotFoundError{}) {
+		return err
+	}
+
+	// Check if the profile hash is unchanged for existing nodes.
+	if s.isProfileHashUnchanged(node, oldNode) {
+		// If we simply return here, the node will remain stuck
+		// in the received status. If we update, then MongoDB's
+		// last_updated and __v will also be updated. What is the
+		// desired behavior in this case? Is it acceptable to update
+		// MongoDB but not update Elasticsearch?
+		return nil
```

Comment:
@geoffturk Could you please have a check?

Reply:
It is acceptable to update the last_updated and version vector fields in MongoDB in this case, provided that the last_updated field in Elasticsearch remains unaltered (to avoid the issue of incorrectly flagging a node as changed when it actually has not). So yes, go ahead and update these two fields in MongoDB in this case, but leave ES unchanged.

One advantage to this approach is that we can monitor the last_updated dates/times and the total number of __v increments to identify nodes that are contacting the index excessively, which will be a useful metric to monitor.

Reply:
ok